Upgrade to Python 3.13

SailingSteve opened the pull request:
Updated the README_MAC_SIMPLIFIED_INSTALL

I did not go forward with removing venv.

Instead of writing out all the scenarios for updating/adding venvs, I added a link to JetBrains well written instructions for doing this.

Was having great difficulty getting the verify_email package going in Python 3.13, and it had not been updated in ten years, so I moved the single file verify_email.py from the verify_email distribution into wevote_functions.

Be sure to update to the latest Pycharm, since older versions use a deprecated "isAlive" attribute, which has been removed from Python 3.12 -- the error is in PyCharm's internal python debugger.

I updated to PyCharm 2024.3.5

Resolves https://wevoteusa.atlassian.net/issues/WV-923

SailingSteve committed:
Upgrade to Python 3.13
Updated the README_MAC_SIMPLIFIED_INSTALL
I did not go forward with removing venv.
Instead of writing out all the scenarios for updating/adding venvs, I added a link to JetBrains well written instructions for doing this.
Was having great difficulty getting the verify_email package going in Python 3.13, and it had not been updated in ten years, so I moved the single file verify_email.py from the verify_email distribution into wevote_functions.
Be sure to update to the latest Pycharm, since older versions use a deprecated "isAlive" attribute, which causes problems with Python 3.12

SailingSteve committed:
Upgrade to Python 3.13
Updated the README_MAC_SIMPLIFIED_INSTALL
I did not go forward with removing venv.
Instead of writing out all the scenarios for updating/adding venvs, I added a link to JetBrains well written instructions for doing this.
Was having great difficulty getting the verify_email package going in Python 3.13, and it had not been updated in ten years, so I moved the single file verify_email.py from the verify_email distribution into wevote_functions.
Be sure to update to the latest Pycharm, since older versions use a deprecated "isAlive" attribute, which causes problems with Python 3.12

SailingSteve committed:
Upgrade to Python 3.13
Updated the README_MAC_SIMPLIFIED_INSTALL
I did not go forward with removing venv.
Instead of writing out all the scenarios for updating/adding venvs, I added a link to JetBrains well written instructions for doing this.
Was having great difficulty getting the verify_email package going in Python 3.13, and it had not been updated in ten years, so I moved the single file verify_email.py from the verify_email distribution into wevote_functions.
Be sure to update to the latest Pycharm, since older versions use a deprecated "isAlive" attribute, which causes problems with Python 3.12

SailingSteve committed:
Upgrade to Python 3.13
Updated the README_MAC_SIMPLIFIED_INSTALL
I did not go forward with removing venv.
Instead of writing out all the scenarios for updating/adding venvs, I added a link to JetBrains well written instructions for doing this.
Was having great difficulty getting the verify_email package going in Python 3.13, and it had not been updated in ten years, so I moved the single file verify_email.py from the verify_email distribution into wevote_functions.
Be sure to update to the latest Pycharm, since older versions use a deprecated "isAlive" attribute, which causes problems with Python 3.12

SailingSteve committed:
Upgrade to Python 3.13
Updated the README_MAC_SIMPLIFIED_INSTALL
I did not go forward with removing venv.
Instead of writing out all the scenarios for updating/adding venvs, I added a link to JetBrains well written instructions for doing this.
Was having great difficulty getting the verify_email package going in Python 3.13, and it had not been updated in ten years, so I moved the single file verify_email.py from the verify_email distribution into wevote_functions.
Be sure to update to the latest Pycharm, since older versions use a deprecated "isAlive" attribute, which causes problems with Python 3.12

SailingSteve committed:
Upgrade to Python 3.13
Updated the README_MAC_SIMPLIFIED_INSTALL
I did not go forward with removing venv.
Instead of writing out all the scenarios for updating/adding venvs, I added a link to JetBrains well written instructions for doing this.
Was having great difficulty getting the verify_email package going in Python 3.13, and it had not been updated in ten years, so I moved the single file verify_email.py from the verify_email distribution into wevote_functions.
Be sure to update to the latest Pycharm, since older versions use a deprecated "isAlive" attribute, which causes problems with Python 3.12

SailingSteve committed:
Upgrade to Python 3.13
Updated the README_MAC_SIMPLIFIED_INSTALL
I did not go forward with removing venv.
Instead of writing out all the scenarios for updating/adding venvs, I added a link to JetBrains well written instructions for doing this.
Was having great difficulty getting the verify_email package going in Python 3.13, and it had not been updated in ten years, so I moved the single file verify_email.py from the verify_email distribution into wevote_functions.
Be sure to update to the latest Pycharm, since older versions use a deprecated "isAlive" attribute, which causes problems with Python 3.12

SailingSteve committed:
Upgrade to Python 3.13
Updated the README_MAC_SIMPLIFIED_INSTALL
I did not go forward with removing venv.
Instead of writing out all the scenarios for updating/adding venvs, I added a link to JetBrains well written instructions for doing this.
Was having great difficulty getting the verify_email package going in Python 3.13, and it had not been updated in ten years, so I moved the single file verify_email.py from the verify_email distribution into wevote_functions.
Be sure to update to the latest Pycharm, since older versions use a deprecated "isAlive" attribute, which causes problems with Python 3.12

DaleMcGrew commented on requirements.txt:
```diff
@@ -29,21 +28,21 @@ google-auth==2.23.4
 oauth2client==4.1.3
 oauthlib==3.2.2
-pandas~=2.1.3           # recommend engine - comment out if initial installation is failing
+pandas~=2.2.3
```
All of the packages labeled "# recommend engine - comment out if initial installation is failing" were added for WeVoteServer/politician/controllers_recommendation.py a machine learning system which was created by a UC Berkeley Masters student. We have had some other engineers run into problems installing the packages required for that system to work, so we keep this label in place so we know what to comment out if a new engineer's installation is blocked.

DaleMcGrew committed:
Update requirements.txt

Added comment back in:
# recommend engine - comment out if initial installation is failing

DaleMcGrew reviewed (approved):
Thank you @SailingSteve! 👍
